### PR TITLE
First implementation of the Bazel toolchain

### PR DESCRIPTION
**Pull request by tapia**

### CODE REVIEW

File: conan/tools/google/bazel.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from conan.tools import CONAN_TOOLCHAIN_ARGS_FILE
+from conans.util.files import load
+
+class Bazel(object):
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._get_bazel_project_configuration()
+
+    def configure(self, args=None):
```
Comment (memsharded) [inline conan/tools/google/bazel.py:12]: If no config step in bazel, you can safely remove this method.

File: conan/tools/google/bazeldeps.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            result += '    static_library = "{0}/lib{1}.a"\n'.format(
+                dependency.new_cpp_info.libdirs[0],
+                lib
+            )
+            result += ')\n\n'
+
+        result += 'cc_library(\n'
+        result += '    name = "{}",\n'.format(dependency.ref.name)
+
+        if dependency.new_cpp_info.includedirs:
```
Comment (memsharded) [inline conan/tools/google/bazeldeps.py]: This might read a bit easier with a jinja template

**Review by memsharded — COMMENTED** [review]
Looking good!

Please start adding some "integration" tests that check things without building anything or calling bazel at all. We will discuss what to do with CI.

File: conan/tools/google/bazeldeps.py (hunk 2 above)
Comment (tapia) [reply conan/tools/google/bazeldeps.py]: Done

File: conan/tools/google/bazel.py (hunk 1 above)
Comment (tapia) [reply conan/tools/google/bazel.py:12]: As far as I've seen, the convention when writing the `build()` method for a project's `conanfile.py` is:

```
def build(self):
    bazel = Bazel(self)
    bazel.configure()
    bazel.build(whatever)
```

Wouldn't deleting the configure method break this pattern?

**Pull request closed without merging**